{"text": "2018-12-17 17:05:49.027 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.004920", "seconds": 0.00492}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67655, "name": "MainProcess"}, "thread": {"id": 139938694395712, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:05:49.027631+08:00", "timestamp": 1545037549.027631}}}
{"text": "2018-12-17 17:05:49.128 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.105515", "seconds": 0.105515}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 67655, "name": "MainProcess"}, "thread": {"id": 139938694395712, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:05:49.128226+08:00", "timestamp": 1545037549.128226}}}
{"text": "2018-12-17 17:05:49.128 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (67655), thread 'MainThread' (139938694395712):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f45f449e9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.105762", "seconds": 0.105762}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (67655), thread 'MainThread' (139938694395712):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67655, "name": "MainProcess"}, "thread": {"id": 139938694395712, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:05:49.128473+08:00", "timestamp": 1545037549.128473}}}
{"text": "2018-12-17 17:05:49.197 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.175198", "seconds": 0.175198}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67655, "name": "MainProcess"}, "thread": {"id": 139938694395712, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:05:49.197909+08:00", "timestamp": 1545037549.197909}}}
{"text": "2018-12-17 17:06:08.817 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.005563", "seconds": 0.005563}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67668, "name": "MainProcess"}, "thread": {"id": 140267524515648, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:06:08.817942+08:00", "timestamp": 1545037568.817942}}}
{"text": "2018-12-17 17:06:08.818 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.005814", "seconds": 0.005814}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 67668, "name": "MainProcess"}, "thread": {"id": 140267524515648, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:06:08.818193+08:00", "timestamp": 1545037568.818193}}}
{"text": "2018-12-17 17:06:08.818 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (67668), thread 'MainThread' (140267524515648):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f928418b9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.005977", "seconds": 0.005977}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (67668), thread 'MainThread' (140267524515648):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67668, "name": "MainProcess"}, "thread": {"id": 140267524515648, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:06:08.818356+08:00", "timestamp": 1545037568.818356}}}
{"text": "2018-12-17 17:06:08.869 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.057467", "seconds": 0.057467}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67668, "name": "MainProcess"}, "thread": {"id": 140267524515648, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:06:08.869846+08:00", "timestamp": 1545037568.869846}}}
{"text": "2018-12-17 17:16:00.319 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.005350", "seconds": 0.00535}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67925, "name": "MainProcess"}, "thread": {"id": 139862895560512, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:16:00.319353+08:00", "timestamp": 1545038160.319353}}}
{"text": "2018-12-17 17:16:00.319 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.005970", "seconds": 0.00597}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 67925, "name": "MainProcess"}, "thread": {"id": 139862895560512, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:16:00.319973+08:00", "timestamp": 1545038160.319973}}}
{"text": "2018-12-17 17:16:00.320 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (67925), thread 'MainThread' (139862895560512):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f344e5259d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.006196", "seconds": 0.006196}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (67925), thread 'MainThread' (139862895560512):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67925, "name": "MainProcess"}, "thread": {"id": 139862895560512, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:16:00.320199+08:00", "timestamp": 1545038160.320199}}}
{"text": "2018-12-17 17:16:00.374 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.060875", "seconds": 0.060875}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67925, "name": "MainProcess"}, "thread": {"id": 139862895560512, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:16:00.374878+08:00", "timestamp": 1545038160.374878}}}
{"text": "2018-12-17 17:17:30.816 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.005099", "seconds": 0.005099}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67965, "name": "MainProcess"}, "thread": {"id": 139869202777920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:17:30.816897+08:00", "timestamp": 1545038250.816897}}}
{"text": "2018-12-17 17:17:30.817 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.005340", "seconds": 0.00534}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 67965, "name": "MainProcess"}, "thread": {"id": 139869202777920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:17:30.817138+08:00", "timestamp": 1545038250.817138}}}
{"text": "2018-12-17 17:17:30.817 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (67965), thread 'MainThread' (139869202777920):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f35c642d9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.005501", "seconds": 0.005501}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (67965), thread 'MainThread' (139869202777920):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67965, "name": "MainProcess"}, "thread": {"id": 139869202777920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:17:30.817299+08:00", "timestamp": 1545038250.817299}}}
{"text": "2018-12-17 17:17:30.870 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.059021", "seconds": 0.059021}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67965, "name": "MainProcess"}, "thread": {"id": 139869202777920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:17:30.870819+08:00", "timestamp": 1545038250.870819}}}
{"text": "2018-12-17 17:17:31.257 | ERROR    | __main__:<module>:74 - An error has been caught in function '<module>', process 'MainProcess' (67965), thread 'MainThread' (139869202777920):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 74, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f35c642d9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.445952", "seconds": 0.445952}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 74, "message": "An error has been caught in function '<module>', process 'MainProcess' (67965), thread 'MainThread' (139869202777920):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67965, "name": "MainProcess"}, "thread": {"id": 139869202777920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:17:31.257750+08:00", "timestamp": 1545038251.25775}}}
{"text": "2018-12-17 17:19:21.267 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.005342", "seconds": 0.005342}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67998, "name": "MainProcess"}, "thread": {"id": 140356792256320, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:21.267585+08:00", "timestamp": 1545038361.267585}}}
{"text": "2018-12-17 17:19:21.267 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.005622", "seconds": 0.005622}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 67998, "name": "MainProcess"}, "thread": {"id": 140356792256320, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:21.267865+08:00", "timestamp": 1545038361.267865}}}
{"text": "2018-12-17 17:19:21.268 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (67998), thread 'MainThread' (140356792256320):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fa74cdbd9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.005783", "seconds": 0.005783}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (67998), thread 'MainThread' (140356792256320):", "module": "loguru_demo", "name": "__main__", "process": {"id": 67998, "name": "MainProcess"}, "thread": {"id": 140356792256320, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:21.268026+08:00", "timestamp": 1545038361.268026}}}
{"text": "2018-12-17 17:19:21.331 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.069418", "seconds": 0.069418}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 67998, "name": "MainProcess"}, "thread": {"id": 140356792256320, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:21.331661+08:00", "timestamp": 1545038361.331661}}}
{"text": "2018-12-17 17:19:33.223 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.004677", "seconds": 0.004677}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68013, "name": "MainProcess"}, "thread": {"id": 140708508215104, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:33.223458+08:00", "timestamp": 1545038373.223458}}}
{"text": "2018-12-17 17:19:33.223 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.004940", "seconds": 0.00494}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 68013, "name": "MainProcess"}, "thread": {"id": 140708508215104, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:33.223721+08:00", "timestamp": 1545038373.223721}}}
{"text": "2018-12-17 17:19:33.223 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (68013), thread 'MainThread' (140708508215104):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7ff930c3c9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.005105", "seconds": 0.005105}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (68013), thread 'MainThread' (140708508215104):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68013, "name": "MainProcess"}, "thread": {"id": 140708508215104, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:33.223886+08:00", "timestamp": 1545038373.223886}}}
{"text": "2018-12-17 17:19:33.278 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.060171", "seconds": 0.060171}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68013, "name": "MainProcess"}, "thread": {"id": 140708508215104, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:33.278952+08:00", "timestamp": 1545038373.278952}}}
{"text": "2018-12-17 17:19:57.589 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.004294", "seconds": 0.004294}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68029, "name": "MainProcess"}, "thread": {"id": 140576847193920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:57.589478+08:00", "timestamp": 1545038397.589478}}}
{"text": "2018-12-17 17:19:57.590 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.004816", "seconds": 0.004816}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 68029, "name": "MainProcess"}, "thread": {"id": 140576847193920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:57.590000+08:00", "timestamp": 1545038397.59}}}
{"text": "2018-12-17 17:19:57.590 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (68029), thread 'MainThread' (140576847193920):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fda8927b9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.005080", "seconds": 0.00508}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (68029), thread 'MainThread' (140576847193920):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68029, "name": "MainProcess"}, "thread": {"id": 140576847193920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:57.590264+08:00", "timestamp": 1545038397.590264}}}
{"text": "2018-12-17 17:19:57.642 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.057154", "seconds": 0.057154}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68029, "name": "MainProcess"}, "thread": {"id": 140576847193920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:57.642338+08:00", "timestamp": 1545038397.642338}}}
{"text": "2018-12-17 17:19:57.796 | ERROR    | __main__:<module>:74 - An error has been caught in function '<module>', process 'MainProcess' (68029), thread 'MainThread' (140576847193920):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 74, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fda8927b9d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.211241", "seconds": 0.211241}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 74, "message": "An error has been caught in function '<module>', process 'MainProcess' (68029), thread 'MainThread' (140576847193920):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68029, "name": "MainProcess"}, "thread": {"id": 140576847193920, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:19:57.796425+08:00", "timestamp": 1545038397.796425}}}
{"text": "2018-12-17 17:21:05.869 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.004514", "seconds": 0.004514}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68048, "name": "MainProcess"}, "thread": {"id": 140339271776064, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:05.869871+08:00", "timestamp": 1545038465.869871}}}
{"text": "2018-12-17 17:21:05.870 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.004775", "seconds": 0.004775}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 68048, "name": "MainProcess"}, "thread": {"id": 140339271776064, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:05.870132+08:00", "timestamp": 1545038465.870132}}}
{"text": "2018-12-17 17:21:05.870 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (68048), thread 'MainThread' (140339271776064):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fa3388e89d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.004950", "seconds": 0.00495}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (68048), thread 'MainThread' (140339271776064):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68048, "name": "MainProcess"}, "thread": {"id": 140339271776064, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:05.870307+08:00", "timestamp": 1545038465.870307}}}
{"text": "2018-12-17 17:21:05.936 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.071057", "seconds": 0.071057}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68048, "name": "MainProcess"}, "thread": {"id": 140339271776064, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:05.936414+08:00", "timestamp": 1545038465.936414}}}
{"text": "2018-12-17 17:21:06.182 | ERROR    | __main__:<module>:74 - An error has been caught in function '<module>', process 'MainProcess' (68048), thread 'MainThread' (140339271776064):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 74, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fa3388e89d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.317292", "seconds": 0.317292}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 74, "message": "An error has been caught in function '<module>', process 'MainProcess' (68048), thread 'MainThread' (140339271776064):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68048, "name": "MainProcess"}, "thread": {"id": 140339271776064, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:06.182649+08:00", "timestamp": 1545038466.182649}}}
{"text": "2018-12-17 17:21:21.387 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.014640", "seconds": 0.01464}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68064, "name": "MainProcess"}, "thread": {"id": 139636799375168, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:21.387285+08:00", "timestamp": 1545038481.387285}}}
{"text": "2018-12-17 17:21:21.388 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.016022", "seconds": 0.016022}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 68064, "name": "MainProcess"}, "thread": {"id": 139636799375168, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:21.388667+08:00", "timestamp": 1545038481.388667}}}
{"text": "2018-12-17 17:21:21.389 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (68064), thread 'MainThread' (139636799375168):\nTraceback (most recent call last):\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1664, in <module>\n    main()\n    \u2514 <function main at 0x7effa873fae8>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1658, in main\n    globals = debugger.run(setup['file'], None, None, is_module)\n    \u2502         \u2502        \u2502   \u2502                          \u2514 False\n    \u2502         \u2502        \u2502   \u2514 {'port': 33817, 'vm_type': None, 'client': '127.0.0.1', 'server': False, 'DEBUG_RECORD_SOCKET_READS': False, 'multiproc': True, ...\n    \u2502         \u2502        \u2514 <bound method PyDB.run of <__main__.PyDB object at 0x7effb9c6e390>>\n    \u2502         \u2514 <__main__.PyDB object at 0x7effb9c6e390>\n    \u2514 <built-in function globals>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1068, in run\n    pydev_imports.execfile(file, globals, locals)  # execute the script\n    \u2502             \u2502        \u2502     \u2502        \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502             \u2502        \u2502     \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502             \u2502        \u2514 '/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py'\n    \u2502             \u2514 <function execfile at 0x7effa9f4c268>\n    \u2514 <module '_pydev_bundle.pydev_imports' from '/home/ginger/pycharm-2018.1.2/helpers/pydev/_pydev_bundle/pydev_imports.py'>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/_pydev_imps/_pydev_execfile.py\", line 18, in execfile\n    exec(compile(contents+\"\\n\", file, 'exec'), glob, loc)\n    \u2502    \u2502       \u2502              \u2502              \u2502     \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502    \u2502       \u2502              \u2502              \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502    \u2502       \u2502              \u2514 '/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py'\n    \u2502    \u2502       \u2514 '\"\"\"\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n\"\"\"\\nimport sys\\n\\nfrom loguru import logger\\n\\nlogger.add(sink=\"fi...\n    \u2502    \u2514 <built-in function compile>\n    \u2514 <built-in function exec>\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7effa86fbd90>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.016665", "seconds": 0.016665}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (68064), thread 'MainThread' (139636799375168):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68064, "name": "MainProcess"}, "thread": {"id": 139636799375168, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:21.389310+08:00", "timestamp": 1545038481.38931}}}
{"text": "2018-12-17 17:21:21.584 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.212079", "seconds": 0.212079}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68064, "name": "MainProcess"}, "thread": {"id": 139636799375168, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:21.584724+08:00", "timestamp": 1545038481.584724}}}
{"text": "2018-12-17 17:21:21.931 | ERROR    | __main__:<module>:74 - An error has been caught in function '<module>', process 'MainProcess' (68064), thread 'MainThread' (139636799375168):\nTraceback (most recent call last):\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1664, in <module>\n    main()\n    \u2514 <function main at 0x7effa873fae8>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1658, in main\n    globals = debugger.run(setup['file'], None, None, is_module)\n    \u2502         \u2502        \u2502   \u2502                          \u2514 False\n    \u2502         \u2502        \u2502   \u2514 {'port': 33817, 'vm_type': None, 'client': '127.0.0.1', 'server': False, 'DEBUG_RECORD_SOCKET_READS': False, 'multiproc': True, ...\n    \u2502         \u2502        \u2514 <bound method PyDB.run of <__main__.PyDB object at 0x7effb9c6e390>>\n    \u2502         \u2514 <__main__.PyDB object at 0x7effb9c6e390>\n    \u2514 <built-in function globals>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/pydevd.py\", line 1068, in run\n    pydev_imports.execfile(file, globals, locals)  # execute the script\n    \u2502             \u2502        \u2502     \u2502        \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502             \u2502        \u2502     \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502             \u2502        \u2514 '/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py'\n    \u2502             \u2514 <function execfile at 0x7effa9f4c268>\n    \u2514 <module '_pydev_bundle.pydev_imports' from '/home/ginger/pycharm-2018.1.2/helpers/pydev/_pydev_bundle/pydev_imports.py'>\n\n  File \"/home/ginger/pycharm-2018.1.2/helpers/pydev/_pydev_imps/_pydev_execfile.py\", line 18, in execfile\n    exec(compile(contents+\"\\n\", file, 'exec'), glob, loc)\n    \u2502    \u2502       \u2502              \u2502              \u2502     \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502    \u2502       \u2502              \u2502              \u2514 {'__name__': '__main__', '__doc__': '\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n', '__package__': None, '__loader_...\n    \u2502    \u2502       \u2502              \u2514 '/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py'\n    \u2502    \u2502       \u2514 '\"\"\"\\nhttps://github.com/Delgan/loguru\\npip3 install loguru\\n\"\"\"\\nimport sys\\n\\nfrom loguru import logger\\n\\nlogger.add(sink=\"fi...\n    \u2502    \u2514 <built-in function compile>\n    \u2514 <built-in function exec>\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 74, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7effa86fbd90>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.558839", "seconds": 0.558839}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 74, "message": "An error has been caught in function '<module>', process 'MainProcess' (68064), thread 'MainThread' (139636799375168):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68064, "name": "MainProcess"}, "thread": {"id": 139636799375168, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:21:21.931484+08:00", "timestamp": 1545038481.931484}}}
{"text": "2018-12-17 17:23:38.750 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.006452", "seconds": 0.006452}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68120, "name": "MainProcess"}, "thread": {"id": 140385077380928, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:23:38.750223+08:00", "timestamp": 1545038618.750223}}}
{"text": "2018-12-17 17:23:38.750 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.006786", "seconds": 0.006786}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 68120, "name": "MainProcess"}, "thread": {"id": 140385077380928, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:23:38.750557+08:00", "timestamp": 1545038618.750557}}}
{"text": "2018-12-17 17:23:38.750 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (68120), thread 'MainThread' (140385077380928):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fade2c899d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.007015", "seconds": 0.007015}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (68120), thread 'MainThread' (140385077380928):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68120, "name": "MainProcess"}, "thread": {"id": 140385077380928, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:23:38.750786+08:00", "timestamp": 1545038618.750786}}}
{"text": "2018-12-17 17:23:38.816 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.072820", "seconds": 0.07282}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 68120, "name": "MainProcess"}, "thread": {"id": 140385077380928, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:23:38.816591+08:00", "timestamp": 1545038618.816591}}}
{"text": "2018-12-17 17:23:39.353 | ERROR    | __main__:<module>:74 - An error has been caught in function '<module>', process 'MainProcess' (68120), thread 'MainThread' (140385077380928):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 74, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7fade2c899d8>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.609993", "seconds": 0.609993}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 74, "message": "An error has been caught in function '<module>', process 'MainProcess' (68120), thread 'MainThread' (140385077380928):", "module": "loguru_demo", "name": "__main__", "process": {"id": 68120, "name": "MainProcess"}, "thread": {"id": 140385077380928, "name": "MainThread"}, "time": {"repr": "2018-12-17 17:23:39.353764+08:00", "timestamp": 1545038619.353764}}}
{"text": "2019-03-29 13:59:33.443 | DEBUG    | __main__:<module>:35 - That's it, beautiful and simple logging!\n", "record": {"elapsed": {"repr": "0:00:00.006746", "seconds": 0.006746}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc1e", "name": "DEBUG", "no": 10}, "line": 35, "message": "That's it, beautiful and simple logging!", "module": "loguru_demo", "name": "__main__", "process": {"id": 15190, "name": "MainProcess"}, "thread": {"id": 139847027140416, "name": "MainThread"}, "time": {"repr": "2019-03-29 13:59:33.443705+08:00", "timestamp": 1553839173.443705}}}
{"text": "2019-03-29 13:59:33.449 | INFO     | __main__:<module>:36 - info\n", "record": {"elapsed": {"repr": "0:00:00.012565", "seconds": 0.012565}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u2139\ufe0f", "name": "INFO", "no": 20}, "line": 36, "message": "info", "module": "loguru_demo", "name": "__main__", "process": {"id": 15190, "name": "MainProcess"}, "thread": {"id": 139847027140416, "name": "MainThread"}, "time": {"repr": "2019-03-29 13:59:33.449524+08:00", "timestamp": 1553839173.449524}}}
{"text": "2019-03-29 13:59:33.450 | ERROR    | __main__:<module>:37 - An error has been caught in function '<module>', process 'MainProcess' (15190), thread 'MainThread' (139847027140416):\nTraceback (most recent call last):\n\n> File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 37, in <module>\n    print(my_function(0, 0, 0))\n    \u2502     \u2514 <function my_function at 0x7f309d333598>\n    \u2514 <built-in function print>\n\n  File \"/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py\", line 21, in my_function\n    return 1 / (x + y + z)\n                \u2502   \u2502   \u2514 0\n                \u2502   \u2514 0\n                \u2514 0\n\nZeroDivisionError: division by zero\n", "record": {"elapsed": {"repr": "0:00:00.013114", "seconds": 0.013114}, "exception": {"type": "ZeroDivisionError", "value": "division by zero", "traceback": true}, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\u274c", "name": "ERROR", "no": 40}, "line": 37, "message": "An error has been caught in function '<module>', process 'MainProcess' (15190), thread 'MainThread' (139847027140416):", "module": "loguru_demo", "name": "__main__", "process": {"id": 15190, "name": "MainProcess"}, "thread": {"id": 139847027140416, "name": "MainThread"}, "time": {"repr": "2019-03-29 13:59:33.450073+08:00", "timestamp": 1553839173.450073}}}
{"text": "2019-03-29 13:59:33.547 | SNAKY    | __main__:<module>:41 - Here we go!\n", "record": {"elapsed": {"repr": "0:00:00.110883", "seconds": 0.110883}, "exception": null, "extra": {}, "file": {"name": "loguru_demo.py", "path": "/home/ginger/Projects/Learning/P_Stack/basic/log_demo/loguru_demo.py"}, "function": "<module>", "level": {"icon": "\ud83d\udc0d", "name": "SNAKY", "no": 38}, "line": 41, "message": "Here we go!", "module": "loguru_demo", "name": "__main__", "process": {"id": 15190, "name": "MainProcess"}, "thread": {"id": 139847027140416, "name": "MainThread"}, "time": {"repr": "2019-03-29 13:59:33.547842+08:00", "timestamp": 1553839173.547842}}}
